aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/io/EventLoopGroupNative.kt

```diff
@@ -16,12 +19,42 @@ import aws.sdk.kotlin.crt.Closeable
  * Otherwise, maxThreads will be the number of event loops in the group.
  * @throws [aws.sdk.kotlin.crt.CrtRuntimeException] If the system is unable to allocate space for a native event loop group
  */
-public actual class EventLoopGroup actual constructor(maxThreads: Int) : Closeable, AsyncShutdown {
+@OptIn(ExperimentalForeignApi::class)
+public actual class EventLoopGroup actual constructor(maxThreads: Int) : CrtResource<aws_event_loop_group>(), Closeable, AsyncShutdown {
+    private val elg: CPointer<aws_event_loop_group>
+
+    override val ptr: CPointer<aws_event_loop_group>
+        get() = elg
```

Comment:
**Question**: I've seen this pattern a few times in the new (to me) Kotlin/Native code. What's the benefit of a separate backing field if the `ptr` always delegates to it anyway?

Reply:
I think we're going to fix this soon. This stems from the original definition of `CrtResource` but I think we're going to just move to something like:

```kotlin
interface NativeHandle<T: CPointed> {
     val ptr: CPointer<T>
}
```

or perhaps name `ptr` to `handle` or `nativeHandle`. Either way I think we'd move to where we don't need a separate backing field.

Reply:
Yes I agree, the backing field sticks out and seems wrong, I plan to refactor after the memory management / `CrtResource` changes are shipped